packages/twenty-front/src/modules/workflow/hooks/useStartNodeCreation.tsx

```diff
@@ -0,0 +1,29 @@
+import { useRightDrawer } from '@/ui/layout/right-drawer/hooks/useRightDrawer';
+import { RightDrawerPages } from '@/ui/layout/right-drawer/types/RightDrawerPages';
+import { workflowCreateStepFromParentStepIdState } from '@/workflow/states/workflowCreateStepFromParentStepIdState';
+import { useCallback } from 'react';
+import { useSetRecoilState } from 'recoil';
+
+export const useStartNodeCreation = () => {
+  const { openRightDrawer } = useRightDrawer();
+  const setWorkflowCreateStepFromParentStepId = useSetRecoilState(
+    workflowCreateStepFromParentStepIdState,
+  );
+
+  /**
+   * This function is used in a context where dependencies shouldn't change much.
+   * That's why its wrapped in a `useCallback` hook.
+   */
```

Comment:
do not put that kind of comment in the code. We try to avoid them.
`The purpose of comments is to explain code that cannot explain itself.` Uncle Bob

Reply:
It's not obvious why I chose to use a `useCallback` here when you first read the code. You might think that it was an error as there is one level of indirection that separates where the function is defined and where it's used.

If the function is not defined with the `useCallback` hook, it will cause bugs where the function is used. And it will be hard to debug.